hoomd/hpmc/ExternalFieldWall.h

```diff
@@ -481,6 +481,86 @@ inline bool test_confined(const CylinderWall& wall,
     return accept;
     }
 
+// Cylindrical Walls and Convex Spheropolyhedra
+DEVICE inline bool test_confined(const CylinderWall& wall,
+                                 const ShapeSpheropolyhedron& shape,
+                                 const vec3<Scalar>& position,
+                                 const vec3<Scalar>& box_origin,
+                                 const BoxDim& box)
+    {
+    bool accept = true;
+    Scalar3 t = vec_to_scalar3(position - box_origin);
+    t.x = t.x - wall.origin.x;
+    t.y = t.y - wall.origin.y;
+    t.z = t.z - wall.origin.z;
+    vec3<Scalar> shifted_pos(box.minImage(t));
+
+    vec3<Scalar> dist_vec
+        = cross(shifted_pos,
+                wall.orientation); // find the component of the shifted position that is
+                                   // perpendicular to the normalized orientation vector
```

Comment:
Comments are easier to read at the same level as the code. You can also use English language rules when writing comments.

```suggestion
    // Find the component of the shifted position that is
    // perpendicular to the normalized orientation vector.
    vec3<Scalar> dist_vec
        = cross(shifted_pos,
                wall.orientation); 
                                   
```